cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts

```diff
@@ -257,3 +258,45 @@ export function filterByCondition(
       };
     });
 }
+
+export function getPluginPropertiesForValidation(nodeInfo: any, widgetJson: IWidgetJson) {
+  const availableProps = new Set();
+  const pluginInfo = Object.assign({}, nodeInfo.plugin);
+  pluginInfo.type = nodeInfo.type;
+  const pluginProperties: PluginProperties = objectQuery(nodeInfo, '_backendProperties');
+  const widgetConfigurationGroup: IConfigurationGroup[] = objectQuery(
+    widgetJson,
+    'configuration-groups'
+  );
+  const widgetOutputs: IWidgetProperty[] = objectQuery(widgetJson, 'outputs');
+  const values: IPropertyValues = objectQuery(nodeInfo, 'plugin', 'properties');
+  const processedConfigurationGroup: IProcessedConfigurationGroups = processConfigurationGroups(
+    pluginProperties,
+    widgetConfigurationGroup,
+    widgetOutputs
+  );
+  const filteredConfigurationGroups: IFilteredConfigurationGroup[] = this.filterByCondition(
```

Comment:
what is `this` refer to? a function will create its own `this` reference. `filterByCondition` is defined in the same file, so you should be able to call it by the function name